### PR TITLE
🗃️#480: Update Types for AssentFormIdentifier Fields

### PR DESCRIPTION
**Pull request by xusandrew**
## Description of Changes

- Add `hasOhip` and `assentFormIdentifier` fields to ParticipantIdentityBase
- Create `vitalStatus` type and add to ClinicalProfile
- Add  `assentFormIdentifier` to InviteParticipantFields
- Update tests

### Special Instructions
```
pnpm build
```

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

**Review by anncatton — COMMENTED** [review]
Good start! Unfortunately there are a couple requirements that didn't occur to me until I started looking at this pr, so I'm leaving a solid chunk of feedback.
I'll update the ticket with the requirement details, but i'll also enumerate them here:

- the `hasOhip` field has a default in the db when the participant record is created, so we don't need to have it as part of the `CreateParticipant` flow. It will be required in the schemas related to updating a participant, as part of the consent wizard implementation. We can leave it in as part of the `ParticipantIdentification` schema as that reflects the logical model.
- the `assentFormIdentifier` is only required for participants with the consentGroup `GUARDIAN_CONSENT_OF_MINOR_INCLUDING_ASSENT`. So we should add refinement logic to ensure this field is populated when needed, and *not* present when not required. This would be similar to how we're handling the guardian vs participant contact info
- the `PICreateParticipantResponse` and `PIClinicianInviteResponse` types need to handle the possibly null value for `assentFormIdentifier`, because it is an optional field. The DB will return null if no value was provided, and this needs to be coerced into `undefined` when the PI DAS service receives the response from Prisma. The subsequent responses from then on do not require any further coercion. An simple example can be found [here](assentFormIdentifier) . There's also some notes about it in the [shared-docs readme](https://github.com/OHCRN/platform/blob/develop/docs/shared-types.md#converting-null-values-to-undefined)
- be sure to test the create participant and clinicianInvite flows end-to-end in swagger to ensure the request and response types are playing nicely!

**Comment by xusandrew** [comment]
@anncatton thank you the thorough review! I have updated this pr with the new changes, and have made sure to test the entire flows

**Review by anncatton — COMMENTED** [review]
Really great work here @xusandrew ! Most of my feedback is just code cleanup and minor enhancements. 😀 

I'd like to split up the conditionalUtils file into smaller files to be put into the `common/utils` folder, but I think that can be a followup chore.

Previously I mentioned adding the `hasOhip` field to the `CreateParticipant` response types but now I think it's not necessary, it's not used directly in the registration flow.

File: packages/types/src/services/piDas/responses/PIClinicianInvite.ts [hunk 1]
```diff
@@ -34,6 +34,10 @@ export const PIClinicianInviteResponse = PIClinicianInvite.extend({ id: NanoId }
 	.extend(GuardianNullableResponseFields)
 	.extend({
 		participantPreferredName: Name.nullable().transform((input) => input ?? undefined),
+		assentFormIdentifier: z
```
Comment (anncatton) [inline packages/types/src/services/piDas/responses/PIClinicianInvite.ts:37]: i missed this before - we should add the transform for `participantEmailAddress` and `participantPhoneNumber` as well

File: apps/consent-ui/src/components/views/Invite/ClinicianInviteForm/index.tsx [hunk 2]
```diff
@@ -58,8 +58,9 @@ const styles = Object.assign({}, formStyles, layoutStyles);
 
 const ClinicianInviteFormRequest = ClinicianInviteBase.extend({
 	participantPreferredName: EmptyOrOptionalName,
-}).refine(hasRequiredInfoForConsentGroup);
-
+})
+	.refine(hasRequiredInfoForConsentGroup)
+	.refine(hasRequiredAssentFormIdentifier);
```
[the remark this answers is not in the document]
Comment (anncatton) [reply apps/consent-ui/src/components/views/Invite/ClinicianInviteForm/index.tsx:63]: Re: the slack convo about whether we'd have the assent form id from object storage at this point - the upload will be handled as a separate action so the id will be available to submit with the invite form 👍

**Review by anncatton — COMMENTED** [review]
Just a couple smol changes then good to go!

**Pull request merged**